### PR TITLE
feat: add `settings.lnbits_running `

### PR DESCRIPTION
**Pull request by motorina0**
https://github.com/lnbits/lnbits/issues/2399

There should be no `while True` loops in the app.
Instead `while settings.lnbits_runing:` should be used.

On the shutdown signal do `settings.lnbits_runing = False`.

This change also helps while testing. Previous infinite loops can no be stoped.

### CODE REVIEW

File: lnbits/app.py [hunk 1]
```diff
@@ -181,7 +182,7 @@ async def check_funding_source() -> None:
     max_retries = settings.funding_source_max_retries
     retry_counter = 0
 
-    while True:
+    while settings.lnbits_runing:
```
Comment (dni) [inline lnbits/app.py:185]:
```suggestion
    while settings.lnbits_running:
```

typo

Comment (motorina0) [reply lnbits/app.py]: fixed, thanks!

**Comment by motorina0** [comment]
@dni does it make sense to set this flag also when `server_restart.set()` is called (`/api/v1/restart` endpoint)?

**Review by talvasconcelos — APPROVED** [review]

**Comment by dni** [comment]
@motorina0 i would set it in `server.py`

**Review by dni — APPROVED** [review]

**Pull request merged**